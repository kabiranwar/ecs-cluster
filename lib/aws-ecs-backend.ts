import * as cdk from '@aws-cdk/core';
import * as ecs from '@aws-cdk/aws-ecs';
import * as ec2 from '@aws-cdk/aws-ec2';
import * as ecs_patterns from "@aws-cdk/aws-ecs-patterns";
import { CfnOutput, CfnParameter } from '@aws-cdk/core';
import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
import * as iam from '@aws-cdk/aws-iam';
import * as ssl_cert from '@aws-cdk/aws-certificatemanager';
import * as route53 from '@aws-cdk/aws-route53';
import { ApplicationLoadBalancedEc2Service, ApplicationMultipleTargetGroupsFargateService, NetworkMultipleTargetGroupsFargateService } from '@aws-cdk/aws-ecs-patterns';
import * as parameters from "../cdk.json"

export class AwsEcsBackEndClusterStack extends cdk.Stack {
  
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const defaultVpc = ec2.Vpc.fromLookup(this, 'ScoutDefaultVpc-backend', {
      isDefault: true,
    });

    const postgrestapiSG = new ec2.SecurityGroup(this, "tdpostgrestdb-SG-BackEnd", {
      vpc: defaultVpc,
      allowAllOutbound: true,
      securityGroupName: parameters.context.config.ProjectName + '-' + parameters.context.config.ProjectEnvironment + '-' + 'postgrestapi-SG',
    });

    postgrestapiSG.addIngressRule(
      ec2.Peer.ipv4('71.163.18.218/32'),
      ec2.Port.tcp(9002),
      "Allow https traffic"
    );
    postgrestapiSG.addIngressRule(
      ec2.Peer.ipv4('128.229.4.0/24'),
      ec2.Port.tcp(9002),
      "Allow https traffic"
    );
    postgrestapiSG.addIngressRule(
      ec2.Peer.ipv4('108.48.153.254/32'),
      ec2.Port.tcp(9002),
      "Allow https traffic"
    );
    postgrestapiSG.addIngressRule(
      ec2.Peer.ipv4('158.80.4.0/24'),
      ec2.Port.tcp(9002),
      "Allow https traffic"
    );
    postgrestapiSG.addIngressRule(
      ec2.Peer.ipv4('128.229.67.14/32'),
      ec2.Port.tcp(9002),
      "Allow https traffic"
    );
    postgrestapiSG.addIngressRule(
      ec2.Peer.ipv4('136.226.18.0/24'),
      ec2.Port.tcp(9002),
      "Allow https traffic"
    );
    //Create an ECS Cluster for back end
    const scoutapi_cluster = ecs.Cluster.fromClusterAttributes(this, 'Imported-Cluster', {
      clusterName: parameters.context.config.ScoutappContainername+parameters.context.config.ProjectEnvironment + parameters.context.config.clustername,
      securityGroups: [ec2.SecurityGroup.fromSecurityGroupId(this, 'IMported SG', 'scout-app-sg')],
      vpc: defaultVpc
    });

    //Task Execution role for Frond end Task Definition
    const ECS_backEnd_TD_Task_Role = iam.Role.fromRoleArn(this, 'imported-TD-Role', `arn:aws:iam::${parameters.context.config.Account_ID}:role/dev-ECS-Front-End-Task-Role`)
    
    //Creation of Execution role for Front end TD:
    const ECS_backEnd_TD_Execution_Role = iam.Role.fromRoleArn(this, 'imported-Execution-Role', `arn:aws:iam::${parameters.context.config.Account_ID}:role/dev-ECS-Front-End-Execution-Role`)
    
  //Imported Security group
  let importedSG = cdk.Fn.importValue("ecssg");

    // Fargate TaskDefinitions 
    const ecssg = ec2.SecurityGroup.fromSecurityGroupId(this, 'imported-sg', importedSG)
    postgrestapiSG.connections.allowFrom(
       ecssg,
       ec2.Port.tcp(9002),
     );
    const tdpostgrestapi = new ecs.FargateTaskDefinition(this, "tdpostgrestapi", {
      family: "tdpostgrestapi",
      memoryLimitMiB: 512,
      cpu: 256,
      taskRole: ECS_backEnd_TD_Task_Role,
      executionRole: ECS_backEnd_TD_Execution_Role
    });
    
    tdpostgrestapi.addContainer('postgrest-api', {
    image: ecs.ContainerImage.fromRegistry('gkoenig/popapp:latest'),
    memoryLimitMiB: 512,
      environment: { PORT: "9002" },    
    //containerName: EnvironmentName.valueAsString + '-Scout-postgrest-Fargate',
    containerName: parameters.context.config.ProjectName+'-'+parameters.context.config.ProjectEnvironment+'-'+'postgrestapi',
    portMappings: [{ containerPort: 9002 }],
    essential: true,
    logging: ecs.LogDriver.awsLogs({ streamPrefix: "scout-postgrestapi-logs" }),
  });       
    // The ECS Service used for deploying tasks 
    const scoutapiservice = new ecs.FargateService(this, "service1-BackEnd", {
      cluster: scoutapi_cluster,
      desiredCount: 1,
      taskDefinition: tdpostgrestapi,
      securityGroups: [postgrestapiSG],
      assignPublicIp: true,
      serviceName: 'scoutapi-service',
    });
    
    // BONUS: Autoscaling based on memory and CPU usage
    const scalableTaget = scoutapiservice.autoScaleTaskCount({
      minCapacity: 1,
      maxCapacity: 2,
    });
    scalableTaget.scaleOnMemoryUtilization("ScaleUpMem-BackEnd", {
      targetUtilizationPercent: 75,
    });
    scalableTaget.scaleOnCpuUtilization("ScaleUpCPU-BackEnd", {
      targetUtilizationPercent: 75,
    });

  }
}
