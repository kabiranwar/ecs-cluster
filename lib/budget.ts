import { App, Construct } from "@aws-cdk/core";
import * as budgets from '@aws-cdk/aws-budgets';
import * as cdk from '@aws-cdk/core';

declare const costFilters: any;
declare const plannedBudgetLimits: any;
// interface BudgetProps {

// }
export class Budget extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);
        // export class Budget extends Construct {
        //     constructor(scope: Construct, id: string, props: BudgetProps) {
        //         super(scope, id);
        new budgets.CfnBudget(this, 'MyCfnBudget', {
            budget: {
                budgetType: 'COST',
                timeUnit: 'MONTHLY ',

                // the properties below are optional
                budgetLimit: {
                    amount: 5,
                    unit: 'USD',
                },
                budgetName: 'Monthly Budget',

            },
            notificationsWithSubscribers: [{
                notification: {
                    comparisonOperator: 'GREATER_THAN ',
                    notificationType: 'ACTUAL ',
                    threshold: 80,
                    thresholdType: 'PERCENTAGE',
                },
                subscribers: [{
                    address: 'anwarkabir2011@gmail.com',
                    subscriptionType: 'EMAIL',
                }]
            }],
        });
    }
}