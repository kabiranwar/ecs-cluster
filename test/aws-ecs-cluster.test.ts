import { expect as expectCDK, matchTemplate, MatchStyle } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as AwsEcsCluster from '../lib/aws-ecs-cluster-stack';

test('Empty Stack', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new AwsEcsCluster.AwsEcsClusterStack(app, 'MyTestStack');
    // THEN
    expectCDK(stack).to(matchTemplate({
      "Resources": {}
    }, MatchStyle.EXACT))
});
