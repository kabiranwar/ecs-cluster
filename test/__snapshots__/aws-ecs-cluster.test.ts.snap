// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app scout stack snapshot 1`] = `
Object {
  "Outputs": Object {
    "ServiceLoadBalancerDNSEC5B149E": Object {
      "Value": Object {
        "Fn::GetAtt": Array [
          "ServiceLBE9A1ADBC",
          "DNSName",
        ],
      },
    },
    "ServiceServiceURL250C0FB6": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "http://",
            Object {
              "Fn::GetAtt": Array [
                "ServiceLBE9A1ADBC",
                "DNSName",
              ],
            },
          ],
        ],
      },
    },
    "cfnoutputALBdevecsserviceALB85354A76": Object {
      "Export": Object {
        "Name": "Scout-ecs-dev-alb",
      },
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "http://",
            Object {
              "Fn::GetAtt": Array [
                "alb8A8B13C2",
                "DNSName",
              ],
            },
          ],
        ],
      },
    },
    "cfnoutputalbsg8CBF59C6": Object {
      "Export": Object {
        "Name": "albsg",
      },
      "Value": "sg-074de470add30d74a",
    },
    "cfnoutputecssg3806F0AA": Object {
      "Export": Object {
        "Name": "ecssg",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "scoutappSG154C1321",
          "GroupId",
        ],
      },
    },
    "cfnoutputimportingalb8A5610FE": Object {
      "Export": Object {
        "Name": "albarn",
      },
      "Value": Object {
        "Ref": "alb8A8B13C2",
      },
    },
  },
  "Parameters": Object {
    "SsmParameterValuePGRSTDBANONROLEC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_DB_ANON_ROLE",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTDBSCHEMAC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_DB_SCHEMA",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTDBURIC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_DB_URI",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTJWTSECRETC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_JWT_SECRET",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePGRSTSECRETISBASE64C96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "PGRST_SECRET_IS_BASE64",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePOSTGRESAUTHROLEC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "POSTGRES_AUTH_ROLE",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePOSTGRESDBC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "POSTGRES_DB",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuePOSTGRESPASSWORDC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "POSTGRES_PASSWORD",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuealbsgC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "albsg",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuemessageC96584B6F00A464EAD1953AFF4B05118Parameter": Object {
      "Default": "message",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "LogGroupTest35764128": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": "/ecs/scout-app-ecs-task-def",
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "ScoutDevClusterA4270C77": Object {
      "Properties": Object {
        "ClusterName": "app-scout-prod-ecs-cluster",
        "ClusterSettings": Array [
          Object {
            "Name": "containerInsights",
            "Value": "enabled",
          },
        ],
      },
      "Type": "AWS::ECS::Cluster",
    },
    "Service9571FDD8": Object {
      "DependsOn": Array [
        "ServiceLBPublicListenerECSGroup0CC8688C",
        "ServiceLBPublicListener46709EAA",
      ],
      "Properties": Object {
        "Cluster": Object {
          "Ref": "ScoutDevClusterA4270C77",
        },
        "DeploymentConfiguration": Object {
          "MaximumPercent": 200,
          "MinimumHealthyPercent": 50,
        },
        "DesiredCount": 1,
        "EnableECSManagedTags": false,
        "HealthCheckGracePeriodSeconds": 60,
        "LaunchType": "FARGATE",
        "LoadBalancers": Array [
          Object {
            "ContainerName": "scout-prod-scout-app",
            "ContainerPort": 8000,
            "TargetGroupArn": Object {
              "Ref": "ServiceLBPublicListenerECSGroup0CC8688C",
            },
          },
        ],
        "NetworkConfiguration": Object {
          "AwsvpcConfiguration": Object {
            "AssignPublicIp": "ENABLED",
            "SecurityGroups": Array [
              Object {
                "Fn::GetAtt": Array [
                  "scoutappSG154C1321",
                  "GroupId",
                ],
              },
            ],
            "Subnets": Array [
              "s-12345",
              "s-67890",
            ],
          },
        },
        "ServiceName": "scout-prod-ecs-service",
        "TaskDefinition": Object {
          "Ref": "TaskDefinitionsimplehttpBE59A3F2",
        },
      },
      "Type": "AWS::ECS::Service",
    },
    "ServiceLBE9A1ADBC": Object {
      "Properties": Object {
        "LoadBalancerAttributes": Array [
          Object {
            "Key": "deletion_protection.enabled",
            "Value": "false",
          },
        ],
        "Scheme": "internet-facing",
        "SecurityGroups": Array [
          Object {
            "Fn::GetAtt": Array [
              "ServiceLBSecurityGroupF7435A5C",
              "GroupId",
            ],
          },
        ],
        "Subnets": Array [
          "s-12345",
          "s-67890",
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "ServiceLBPublicListener46709EAA": Object {
      "Properties": Object {
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "ServiceLBPublicListenerECSGroup0CC8688C",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "ServiceLBE9A1ADBC",
        },
        "Port": 80,
        "Protocol": "HTTP",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "ServiceLBPublicListenerECSGroup0CC8688C": Object {
      "Properties": Object {
        "Port": 80,
        "Protocol": "HTTP",
        "TargetType": "ip",
        "VpcId": "vpc-12345",
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "ServiceLBSecurityGroupF7435A5C": Object {
      "Properties": Object {
        "GroupDescription": "Automatically created Security Group for ELB ECSClusterTestStackServiceLBF71FD3FD",
        "SecurityGroupIngress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow from anyone on port 80",
            "FromPort": 80,
            "IpProtocol": "tcp",
            "ToPort": 80,
          },
        ],
        "VpcId": "vpc-12345",
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "ServiceLBSecurityGrouptoECSClusterTestStackscoutappSGFE16C3B680000EDFC4A8": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "FromPort": 8000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "ServiceLBSecurityGroupF7435A5C",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 8000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ServiceTaskCountTargetCpuScaling4BCDA6AE": Object {
      "Properties": Object {
        "PolicyName": "ECSClusterTestStackServiceTaskCountTargetCpuScaling8739A3E9",
        "PolicyType": "TargetTrackingScaling",
        "ScalingTargetId": Object {
          "Ref": "ServiceTaskCountTargetDF7542A8",
        },
        "TargetTrackingScalingPolicyConfiguration": Object {
          "PredefinedMetricSpecification": Object {
            "PredefinedMetricType": "ECSServiceAverageCPUUtilization",
          },
          "TargetValue": 50,
        },
      },
      "Type": "AWS::ApplicationAutoScaling::ScalingPolicy",
    },
    "ServiceTaskCountTargetDF7542A8": Object {
      "Properties": Object {
        "MaxCapacity": 20,
        "MinCapacity": 1,
        "ResourceId": Object {
          "Fn::Join": Array [
            "",
            Array [
              "service/",
              Object {
                "Ref": "ScoutDevClusterA4270C77",
              },
              "/",
              Object {
                "Fn::GetAtt": Array [
                  "Service9571FDD8",
                  "Name",
                ],
              },
            ],
          ],
        },
        "RoleARN": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::321325872726:role/aws-service-role/ecs.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_ECSService",
            ],
          ],
        },
        "ScalableDimension": "ecs:service:DesiredCount",
        "ServiceNamespace": "ecs",
      },
      "Type": "AWS::ApplicationAutoScaling::ScalableTarget",
    },
    "ServiceTaskCountTargetMemoryScaling502AB098": Object {
      "Properties": Object {
        "PolicyName": "ECSClusterTestStackServiceTaskCountTargetMemoryScalingCBEEA7CF",
        "PolicyType": "TargetTrackingScaling",
        "ScalingTargetId": Object {
          "Ref": "ServiceTaskCountTargetDF7542A8",
        },
        "TargetTrackingScalingPolicyConfiguration": Object {
          "PredefinedMetricSpecification": Object {
            "PredefinedMetricType": "ECSServiceAverageMemoryUtilization",
          },
          "TargetValue": 50,
        },
      },
      "Type": "AWS::ApplicationAutoScaling::ScalingPolicy",
    },
    "TaskDefinitionsimplehttpBE59A3F2": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Essential": true,
            "Image": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "321325872726.dkr.ecr.us-east-1.",
                  Object {
                    "Ref": "AWS::URLSuffix",
                  },
                  "/ecr-simplehttp:1.0",
                ],
              ],
            },
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "LogGroupTest35764128",
                },
                "awslogs-region": "us-east-1",
                "awslogs-stream-prefix": "ecs",
              },
            },
            "Memory": 512,
            "Name": "scout-prod-scout-app",
            "PortMappings": Array [
              Object {
                "ContainerPort": 8000,
                "HostPort": 8000,
                "Protocol": "tcp",
              },
            ],
          },
          Object {
            "Cpu": 256,
            "Environment": Array [
              Object {
                "Name": "message",
                "Value": Object {
                  "Ref": "SsmParameterValuemessageC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_DB_ANON_ROLE",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTDBANONROLEC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_DB_SCHEMA",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTDBSCHEMAC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_DB_URI",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTDBURIC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_JWT_SECRET",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTJWTSECRETC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "PGRST_SECRET_IS_BASE64",
                "Value": Object {
                  "Ref": "SsmParameterValuePGRSTSECRETISBASE64C96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "POSTGRES_AUTH_ROLE",
                "Value": Object {
                  "Ref": "SsmParameterValuePOSTGRESAUTHROLEC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "POSTGRES_DB",
                "Value": Object {
                  "Ref": "SsmParameterValuePOSTGRESDBC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
              Object {
                "Name": "POSTGRES_PASSWORD",
                "Value": Object {
                  "Ref": "SsmParameterValuePOSTGRESPASSWORDC96584B6F00A464EAD1953AFF4B05118Parameter",
                },
              },
            ],
            "Essential": true,
            "Image": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "321325872726.dkr.ecr.us-east-1.",
                  Object {
                    "Ref": "AWS::URLSuffix",
                  },
                  "/ecr-metalapp:latest",
                ],
              ],
            },
            "LogConfiguration": Object {
              "LogDriver": "awslogs",
              "Options": Object {
                "awslogs-group": Object {
                  "Ref": "LogGroupTest35764128",
                },
                "awslogs-region": "us-east-1",
                "awslogs-stream-prefix": "ecs",
              },
            },
            "Memory": 512,
            "Name": "scout-prod-postgrest-api",
            "PortMappings": Array [
              Object {
                "ContainerPort": 3000,
                "HostPort": 3000,
                "Protocol": "tcp",
              },
            ],
          },
        ],
        "Cpu": "512",
        "ExecutionRoleArn": "arn:aws:iam::321325872726:role/ecsTaskExecutionRole",
        "Family": "scout-prod-ecs-task-def",
        "Memory": "1024",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "TaskRoleArn": "arn:aws:iam::321325872726:role/ecsTaskExecutionRole",
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "alb8A8B13C2": Object {
      "Properties": Object {
        "LoadBalancerAttributes": Array [
          Object {
            "Key": "deletion_protection.enabled",
            "Value": "false",
          },
        ],
        "Name": "app-scout-prod-alb",
        "Scheme": "internet-facing",
        "SecurityGroups": Array [
          "sg-074de470add30d74a",
        ],
        "Subnets": Array [
          "s-12345",
          "s-67890",
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "albalbhttplistenerE2CBF17B": Object {
      "Properties": Object {
        "DefaultActions": Array [
          Object {
            "TargetGroupArn": Object {
              "Ref": "scouthttptargetgroup5C634259",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": Object {
          "Ref": "alb8A8B13C2",
        },
        "Port": 80,
        "Protocol": "HTTP",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "scoutappSG154C1321": Object {
      "Properties": Object {
        "GroupDescription": "Default/ECSClusterTestStack/scoutapp-SG/scoutapp-SG",
        "GroupName": "scout-app-sg",
        "VpcId": "vpc-12345",
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "scoutappSGfrom1084815325432300075918113": Object {
      "Properties": Object {
        "CidrIp": "108.48.153.254/32",
        "Description": "Allow http traffic",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom1084815325432801ABFADE5": Object {
      "Properties": Object {
        "CidrIp": "108.48.153.254/32",
        "Description": "Allow http traffic",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom128229402430001DE6F79A": Object {
      "Properties": Object {
        "CidrIp": "128.229.4.0/24",
        "Description": "Allow http traffic",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom1282294024806C7F4938": Object {
      "Properties": Object {
        "CidrIp": "128.229.4.0/24",
        "Description": "Allow http traffic",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom1282296714323000EC3A1058": Object {
      "Properties": Object {
        "CidrIp": "128.229.67.14/32",
        "Description": "Allow http traffic",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom1282296714328000BA415C": Object {
      "Properties": Object {
        "CidrIp": "128.229.67.14/32",
        "Description": "Allow http traffic",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom136226180243000DD6A5D92": Object {
      "Properties": Object {
        "CidrIp": "136.226.18.0/24",
        "Description": "Allow http traffic",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom136226180248062ABBFCD": Object {
      "Properties": Object {
        "CidrIp": "136.226.18.0/24",
        "Description": "Allow http traffic",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom158804024300043358F3A": Object {
      "Properties": Object {
        "CidrIp": "158.80.4.0/24",
        "Description": "Allow http traffic",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom15880402480ADB20027": Object {
      "Properties": Object {
        "CidrIp": "158.80.4.0/24",
        "Description": "Allow http traffic",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom172310016300026EE24B0": Object {
      "Properties": Object {
        "CidrIp": "172.31.0.0/16",
        "Description": "Allow http traffic",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom172310016808394A816": Object {
      "Properties": Object {
        "CidrIp": "172.31.0.0/16",
        "Description": "Allow http traffic",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom7116318218323000221154CF": Object {
      "Properties": Object {
        "CidrIp": "71.163.18.218/32",
        "Description": "Allow http traffic",
        "FromPort": 3000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 3000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfrom71163182183280932EAD11": Object {
      "Properties": Object {
        "CidrIp": "71.163.18.218/32",
        "Description": "Allow http traffic",
        "FromPort": 80,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 80,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfromECSClusterTestStackSGA0EE9013ALLPORTS203B4C0C": Object {
      "Properties": Object {
        "Description": "from ECSClusterTestStackSGA0EE9013:ALL PORTS",
        "FromPort": 0,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": "sg-074de470add30d74a",
        "ToPort": 65535,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGfromECSClusterTestStackServiceLBSecurityGroup2C71137F8000AED5C9EF": Object {
      "Properties": Object {
        "Description": "Load balancer to target",
        "FromPort": 8000,
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": Object {
          "Fn::GetAtt": Array [
            "ServiceLBSecurityGroupF7435A5C",
            "GroupId",
          ],
        },
        "ToPort": 8000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "scoutappSGto00000ALLTRAFFICD53597B5": Object {
      "Properties": Object {
        "CidrIp": "0.0.0.0/0",
        "Description": "Allow all outbound traffic by default",
        "GroupId": Object {
          "Fn::GetAtt": Array [
            "scoutappSG154C1321",
            "GroupId",
          ],
        },
        "IpProtocol": "-1",
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "scouthttptargetgroup5C634259": Object {
      "Properties": Object {
        "HealthCheckPath": "/filters",
        "HealthCheckProtocol": "HTTP",
        "Name": "app-scout-prod-tg",
        "Port": 80,
        "Protocol": "HTTP",
        "TargetType": "ip",
        "VpcId": "vpc-12345",
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
  },
}
`;
